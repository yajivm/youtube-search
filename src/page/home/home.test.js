import React from 'react';
import { render, waitFor } from '@testing-library/react';

import Home from './home';

const mockScrollReachedBottom = jest.fn();
jest.mock('../../helpers/utils', () => {
    return {
        scrollReachedBottom: () => mockScrollReachedBottom(),
    }
});

jest.mock('../../components/Header', () => (props) => <div data-testid="header" {...props} />);
jest.mock('../../components/VideosList', () => (props) => <div data-testid="video-list" {...props} />);
jest.mock('../../components/VideoPlayer', () => (props) => <div data-testid="video-player" {...props} />);

const mockResponseData = {
    items: [
        {
            id: {
                videoId: 'aTestIdOne',
            },
            snippet: {
                publishedAt: "2021-11-12T22:18:07Z",
                title: "aTestTitleOne",
                channelId: "aTestChannelIdOne",
                thumbnails: {
                    high: {
                        url: "aTestUrlOne",
                    }
                },
            }
        },
        {
            id: {
                videoId: 'aTestIdTwo',
            },
            snippet: {
                publishedAt: "2021-11-12T22:18:07Z",
                title: "aTestTitleTwo",
                channelId: "aTestChannelIdTwo",
                thumbnails: {
                    high: {
                        url: "aTestUrlTwo",
                    }
                },
            }
        }
    ]
}

const mockGetVideoDetailsByKey = jest.fn();
jest.mock('../../services/youtube.service', () => {
    return {
        getVideoDetailsByKey: (...args) => mockGetVideoDetailsByKey(...args),
    }
});

test('should render header component', async () => {
    mockGetVideoDetailsByKey.mockResolvedValueOnce({ data: mockResponseData });
    const { getByTestId } = await render(<Home />);

    const headerElement = getByTestId('header');

    expect(headerElement).toBeDefined();
});

test('should render video list component', async () => {
    mockGetVideoDetailsByKey.mockReset();
    mockGetVideoDetailsByKey.mockResolvedValueOnce({ data: mockResponseData });
    const { getByTestId } = render(<Home />);

    const videoListElement = getByTestId('video-list');

    expect(mockGetVideoDetailsByKey).toHaveBeenCalledTimes(1);
    expect(mockGetVideoDetailsByKey).toHaveBeenCalledWith('youtube', 30);
    expect(videoListElement).toBeDefined();
    expect(videoListElement).toMatchSnapshot();
});