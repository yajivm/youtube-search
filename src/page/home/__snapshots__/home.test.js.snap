// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render video list component 1`] = `
<div
  data-testid="video-list"
  listdata=""
/>
`;
